platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClient.java

```diff
@@ -161,50 +159,21 @@ private Element createSoapMessage(AttributeQuery attributeQuery)
     }
 
     /**
-     * Sends the request to the external attribute store via an HTTPS Client.
+     * Retrieves the response and returns the SAML Assertion.
      *
      * @param requestDocument of the request.
-     * @return Assertion of the response or null if the response contains a bad status code.
+     * @return Assertion of the response or null if the response is empty.
      * @throws AttributeQueryException
      */
-    private Assertion sendRequest(Document requestDocument) throws AttributeQueryException {
-        URL url;
-        HttpURLConnection connection;
-        try {
-            url = new URL(externalAttributeStoreUrl);
-
-            connection = createHttpsUrlConnection(url);
-            connection.setRequestMethod("POST");
-            connection.setRequestProperty("Content-Type", "text/xml");
-            connection.setRequestProperty("SOAPAction", SOAP_ACTION);
-            connection.setRequestProperty("Cache-Control", "no-cache, no-store");
-            connection.setRequestProperty("Pragma", "no-cache");
-            connection.setDoOutput(true);
-
-            TransformerProperties transformerProperties = new TransformerProperties();
-            transformerProperties.addOutputProperty(OutputKeys.OMIT_XML_DECLARATION, "yes");
-
-            try (DataOutputStream write = new DataOutputStream(connection.getOutputStream())) {
-                write.writeBytes(XMLUtils.format(requestDocument, transformerProperties));
-                write.flush();
-
-                if (!reportStatusCode(connection.getResponseCode())) {
-                    // If connection is unsuccessful, do not continue and return null.
-                    return null;
-                }
-            }
-        } catch (IOException e) {
-            LOGGER.error("Error when creating connection, unable to send request to {}",
-                    externalAttributeStoreUrl);
-            throw new AttributeQueryException("Could not create connection.", e);
-        }
-
+    private Assertion retrieveResponse(Document requestDocument) throws AttributeQueryException {
         try {
             Assertion assertion = null;
-            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
-            documentBuilderFactory.setNamespaceAware(true);
-            Document responseDocument = documentBuilderFactory.newDocumentBuilder()
-                    .parse(connection.getInputStream());
+            Document responseDocument = (responseDocument = sendRequest(requestDocument)) == null ?
```

Comment:
This line is very confusing.  I would suggest avoiding variable shadowing.  I also do not see what the ternary is doing.